client/src/components/Profile/EditAvailability.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import { TextField, Grid, Button, Checkbox, Switch } from '@material-ui/core';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import { UserProfile } from '../../interface/Profile';
+import { Formik, FormikHelpers } from 'formik';
+import moment from 'moment';
+import useStyles from './useStyles';
+
+interface Props {
+  currentUserProfile: UserProfile;
+  handleProfileSubmit: (
+    { availability, isAvailable }: UserProfile,
+    { setSubmitting }: FormikHelpers<UserProfile>,
+  ) => void;
+}
+
+const EditAvailability = ({ handleProfileSubmit, currentUserProfile }: Props): JSX.Element => {
+  const classes = useStyles();
+
+  return (
+    <div>
+      <h1 className={classes.profileTitle}>Edit Availability</h1>
+      <Formik
+        initialValues={{
+          isAvailable: currentUserProfile.isAvailable || false,
+          availability: currentUserProfile?.availability,
+        }}
+        onSubmit={handleProfileSubmit}
+      >
+        {({ handleSubmit, handleChange, setFieldValue, values, isSubmitting }) => (
+          <form onSubmit={handleSubmit}>
+            <Grid container spacing={1} className={classes.form}>
+              <Grid item xs={12}>
+                <div className={classes.textFieldRow}>
```

Comment:
We probably could utilise a box here to work with Box more from Material UI.

Reply:
Use Box to replace the div?